src/jsonld/localBusiness.jsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Head from 'next/head';
+
+import markup from '../utils/markup';
+
+const buildImages = images =>
+  images.length ? `"image": [${images.map(image => `"${image}"`)}]` : '';
+
+const buildGeo = geo => `
+  "geo": {
+    "@type": "GeoCoordinates",
+    "latitude": "${geo.latitude}",
+    "longitude": "${geo.longitude}"
+  },
+`;
+
+const buildAddress = address => `
+  "address": {
+    "@type": "PostalAddress",
+    "streetAddress": "${address.streetAddress}",
+    "addressLocality": "${address.addressLocality}",
+    ${
+      address.addressRegion
+        ? `"addressRegion": "${address.addressRegion}",`
+        : ''
+    }
+    "postalCode": "${address.postalCode}",
+    "addressCountry": "${address.addressCountry}"
+  },
+`;
+
+const LocalBusinessJsonLd = ({
+  type,
+  id,
+  name,
+  description,
+  url,
+  telephone,
+  address,
+  geo,
+  images,
+}) => {
+  // NOTE: Images has no trailing , as it would make the content invalid
+  // If you add more properties update buildImages
+  const jslonld = `{
+    "@context": "http://schema.org",
+    "@type": "${type}",
+    "@id": "${id}",
+    "name": "${name}",
+    ${description ? `"description": "${description}",` : ''}
+    ${url ? `"url": "${url}",` : ''}
+    ${telephone ? `"telephone": "${telephone}",` : ''}
+    ${buildAddress(address)}
+    ${geo ? `${buildGeo(geo)}` : ''}
+    ${buildImages(images)} 
+  }`;
+
+  return (
+    <Head>
+      <script
+        type="application/ld+json"
+        dangerouslySetInnerHTML={markup(jslonld)}
+        key="jsonld-local-business"
+      />
+    </Head>
+  );
+};
+
+LocalBusinessJsonLd.defaultProps = {
+  type: 'LocalBusiness',
+  description: null,
+  url: null,
+  telephone: null,
+  images: [],
+  geo: null,
+};
+
+LocalBusinessJsonLd.propTypes = {
+  type: PropTypes.string,
+  id: PropTypes.string.isRequired,
+  name: PropTypes.string.isRequired,
+  description: PropTypes.string,
+  url: PropTypes.string,
+  telephone: PropTypes.string,
+  address: PropTypes.shape({
+    streetAddress: PropTypes.string.isRequired,
+    addressLocality: PropTypes.string.isRequired,
+    addressRegion: PropTypes.string,
+    postalCode: PropTypes.string.isRequired,
+    addressCountry: PropTypes.string.isRequired,
+  }).isRequired,
+  geo: PropTypes.shape({
+    latitude: PropTypes.string.isRequired,
+    longitude: PropTypes.string.isRequired,
+  }),
+  images: PropTypes.arrayOf(PropTypes.string),
```

Comment:
This should be required?

Reply:
According to the google documentation these aren't required for the base type [LocalBusiness](https://developers.google.com/search/docs/data-types/local-business#local_business_properties). 

However these are required for other types like [Restaurant](https://developers.google.com/search/docs/data-types/local-business#restaurant-properties). 

Looking at (schema.org)[https://schema.org/LocalBusiness] images are valid because it's inherited from Thing. 

Based on this I didn't make them required however it's unclear if google will use them for other business types although I interpreted the docs as them being used in the business panel in search results. 

If anything I think they should be removed not made required. Thoughts?

Reply:
Interesting 🤔 

When checking something like the below on [Google's Structured Data Testing Tool](https://search.google.com/structured-data/testing-tool)

```
{
    "@context": "http://schema.org",
    "@type": "Store",
    "@id": "http://davesdeptstore.example.com",
    "name": "Dave's Department Store",
    "description": "Dave's latest department store in San Jose, now open",
    "url": "http://www.example.com/store-locator/sl/San-Jose-Westgate-Store/1427",
    "telephone": "+14088717984",
  "address": {
    "@type": "PostalAddress",
    "streetAddress": "1600 Saratoga Ave",
    "addressLocality": "San Jose",
    "addressRegion": "CA",
    "postalCode": "95129",
    "addressCountry": "US"
  },
  "geo": {
    "@type": "GeoCoordinates",
    "latitude": "37.293058",
    "longitude": "-121.988331"
  }     
}
```

It throws an error saying images are required. Not sure which one to trust in this case. 

What do you think? Maybe we add a note to the docs and let the dev decide? 

If we are allowing this to be not required, could we move a required field to be last (`name` maybe), this will ensure the JSON does not have a trailing slash in certain scenarios.